keras_nlp/src/models/stable_diffusion_3/stable_diffusion_3_text_to_image.py

```diff
@@ -0,0 +1,244 @@
+            negative_token_ids
+        )
+
+        # Concatenation for classifier-free guidance.
+        embeddings = ops.concatenate(
+            [positive_embeddings, negative_embeddings], axis=0
+        )
+        pooled_embeddings = ops.concatenate(
+            [positive_pooled_embeddings, negative_pooled_embeddings], axis=0
+        )
+        return embeddings, pooled_embeddings
+
+    def denoise_step(
+        self,
+        latents,
+        embeddings,
+        steps,
+        num_steps,
+        classifier_free_guidance_scale,
+    ):
+        contexts, pooled_projections = embeddings
+        sigma = self.backbone.noise_scheduler.get_sigma(steps, num_steps)
+        sigma_next = self.backbone.noise_scheduler.get_sigma(
+            steps + 1, num_steps
+        )
+
+        # Sigma to timestep.
+        timestep = self.backbone.noise_scheduler.sigma_to_timestep(sigma)
+        timestep = ops.broadcast_to(timestep, ops.shape(latents)[:1])
+
+        # Diffusion.
+        predicted_noise = self.backbone.mmdit(
+            {
+                "latent": ops.concatenate([latents, latents], axis=0),
+                "context": contexts,
+                "pooled_projection": pooled_projections,
+                "timestep": ops.concatenate([timestep, timestep], axis=0),
+            },
+            training=False,
+        )
+        predicted_noise = ops.cast(predicted_noise, "float32")
+
+        # Classifier-free guidance.
+        classifier_free_guidance_scale = ops.cast(
+            classifier_free_guidance_scale, predicted_noise.dtype
+        )
+        positive_noise, negative_noise = ops.split(predicted_noise, 2, axis=0)
+        predicted_noise = negative_noise + classifier_free_guidance_scale * (
+            positive_noise - negative_noise
+        )
+
+        # Euler step.
+        latents = self.backbone.noise_scheduler.step(
+            latents, predicted_noise, sigma, sigma_next
+        )
+        return latents
+
+    def decode_step(self, latents):
+        # Latent calibration.
+        latents = ops.add(ops.divide(latents, 1.5305), 0.0609)
+
+        # Decoding.
+        outputs = self.backbone.vae(latents, training=False)
+        return outputs
+
+    def text_to_image_step(
+        self,
+        latents,
+        token_ids,
+        negative_token_ids,
+        num_steps,
+        classifier_free_guidance_scale,
+    ):
+        # Encode inputs.
+        embeddings = self.encode_step(token_ids, negative_token_ids)
+
+        # Denoise.
+        def body_fun(step, latents):
+            latents = self.denoise_step(
+                latents,
+                embeddings,
+                step,
+                num_steps,
+                classifier_free_guidance_scale,
+            )
+            return latents
+
+        latents = ops.fori_loop(0, num_steps, body_fun, latents)
+
+        # Decode.
+        outputs = self.decode_step(latents)
+        return outputs
+
+    def text_to_image(
```

Comment:
What do we think about naming this `generate()`. Just chatted a bit with Francois about this.

Then we have this as a really high level usage...

```python
text_to_image = keras_hub.models.TextToImage.from_preset(
    "stable_diffusion_3_medium"
)
text_to_image.generate("A picture of ...")
```

The exact signature of `generate()` will depend on your task, but I still think it is nice to have that as a common entry point for our generative tasks. `task.generate(input)` as the basic signature.

Reply:
Sure, this is more consistent with the other tasks. Updated.